Angular owner creation

binq1000 opened the pull request:
Registration of an Owner.

binq1000 committed:
Added registration front end.

app.module: Added RegistrationComponent
app.routing: Added /register.
constants: added URL for administration.
http service: Added custom post for when you need to set a body and options.
Owner: recreated restOwner object.
user.service: added register method.

binq1000 committed:
Added routerlink in header.

Added multilanguage.
Added redirecting to homepage after Registering

Added translations.

binq1000 committed:
Merge branch 'development' into Angular_OwnerCreation

nino-vrijman commented on RoadPricingAngular/src/app/http.service.ts:
```diff
@@ -26,6 +26,11 @@ export class HttpService {
     });
   }
 
+  public customPost(url, body, options) {
```
"custom" zegt niks over wat de functie doet, je kan de bestaande post methode aanpassen om een options object te ondersteunen (extra parameter)

binq1000 replied:
Naam verandert, aangezien ik een post wil doen zonder de authorization header.

nino-vrijman commented on RoadPricingAngular/src/app/registration/registration.component.html:
```diff
@@ -0,0 +1,35 @@
+    <form #userForm="ngForm">
+      <div class="form-group">
+        <label for="name">{{ 'REGISTER.NAME_LABEL' | translate }}</label>
+        <input [(ngModel)]="owner.name" id="name" type="text" class="form-control input-sm" name="name" required/>
```
De input velden hoeven niet two-way gebind te worden, alleen one-way zodat de dingen die je typt in de invoervelden in owner property van je component worden opgeslagen.

binq1000 replied:
Resulteerde in erg veel null, niet waard om hier veel tijd nog in te steken.

nino-vrijman commented on RoadPricingAngular/src/app/user.service.ts:
```diff
@@ -43,4 +45,26 @@ export class OwnerService {
+    //return this.httpService.post(url);
+
+    return this.httpService
+      .customPost(url, body, options)
```
Veranderen zodat ie de gewone post methode gebruikt, extractdata geeft een error als je een account dat al bestaat probeert te registreren.

nino-vrijman commented on RoadPricingAngular/src/app/user.service.ts:
```diff
@@ -43,4 +45,26 @@ export class OwnerService {
+   * @param owner The newly created owner
+   */
+  register(owner: Owner) {
+    //Send request to the administration backend. /owner/create with owner in the body.
```
Comment klopt niet meer / is niet recent net zoals de JSdoc hierboven, account / owner wordt door elkaar gebruikt.

binq1000 committed:
Merge branch 'Angular_OwnerCreation' of E:\School\S63B\RoadPricing\RoadPricing with conflicts.

binq1000 committed:
Merge remote-tracking branch 'origin/Angular_OwnerCreation' into Angular_OwnerCreation

binq1000 committed:
User.service.ts: Removed comment. return the status of the call.
http.service.ts: Renamed method. this method should be called when no authentication headers are required.
registration.component.ts: Check the status of the call. When status is not 200 (OK) create the new owner.
